verification/certifyKey.go

```diff
@@ -483,114 +83,259 @@ func testCertifyKey(d TestDPEInstance, c DPEClient, t *testing.T, simulation boo
+		childTCI,
+		DeriveChildFlags(InputAllowX509|RetainParent),
+		tciType,
+		0)
 
-	// Root certificate is expected to be in the beginning of the chain, the rest are expected to be intermediates.
-	roots.AddCert(certChain[0])
+	if err != nil {
+		t.Fatalf("[FATAL]: Error while creating child handle: %s", err)
+	}
 
-	for _, cert := range certChain[1:] {
-		if cert.Subject.String() == cert.Issuer.String() {
-			t.Errorf("[ERROR]: Found a self-signed certificate in middle of certificate chain returned by GetCertificateChain.")
-			continue
+	childHandle := &childCtx.NewContextHandle
+	parentHandle = &childCtx.ParentContextHandle
+
+	// Clean up contexts
+	defer func() {
+		err := c.DestroyContext(childHandle, DestroyDescendants)
+		if err != nil {
+			t.Errorf("[ERROR]: Error while cleaning up derived context, this may cause failure in subsequent tests: %s", err)
 		}
-		intermediates.AddCert(cert)
+	}()
+
+	var childTcbInfo DiceTcbInfo
+	childHandle, childTcbInfo, err = getTcbInfoForHandle(c, childHandle)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not get TcbInfo: %v", err)
+	}
+
+	// Check vendorInfo field in multitcb
+	if err = checkDiceTcbVendorInfo(childTcbInfo, d.GetLocality()); err != nil {
+		t.Errorf("[ERROR]: %v", err)
+	}
+
+	// Check tci type field in multitcb
+	if err = checkCurrentDiceTcbTciType(childTcbInfo, tciType); err != nil {
+		t.Errorf("[ERROR]: %v", err)
+	}
+
+	// Check hash algorithm field in multitcb
+	if err = checkDiceTcbHashAlgorithm(childTcbInfo, hashAlg); err != nil {
+		t.Errorf("[ERROR]: %v", err)
 	}
-	opts := x509.VerifyOptions{
-		Roots:         roots,
-		Intermediates: intermediates,
-		CurrentTime:   time.Now().UTC(),
-		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+
+	// Check cumulative value of derived TCi node
+	// Get last cumulative TCI value
+	// Check dice tcb measurements of child TCI node
+	parentHandle, parentTcbInfo, err := getTcbInfoForHandle(c, parentHandle)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not get parent TCB data. %v", err)
+	}
+	lastCumulativeTCI := parentTcbInfo.Fwids[1].Digest
+	if err = verifyDiceTcbDigest(childTcbInfo, childTCI, lastCumulativeTCI); err != nil {
+		t.Errorf("[ERROR]: %v", err)
 	}
 
-	return opts
+	// Check all dice tcb measurements
+	certifiedKey, err := c.CertifyKey(childHandle, childTCI, CertifyKeyX509, 0)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not certify key: %s", err)
+	}
+	childHandle = &certifiedKey.Handle
+	if err = validateDiceTcbFwids(certifiedKey.Certificate, childTCI); err != nil {
+		t.Errorf("[ERROR]: %v", err)
+	}
 }
 
-// Gets KeyUsage bitmap and returns as list of KeyUsage name strings.
-func getKeyUsageNames(keyUsage x509.KeyUsage) []string {
-	keyUsageNames := []string{}
+// Checks Multi Tcb Info for context derived from simulation mode
+func checkWithDerivedChildContextSimulation(d TestDPEInstance, c DPEClient, t *testing.T, handle *ContextHandle) {
+	profile, err := GetTransportProfile(d)
```

Comment:
New method after adding DeriveContext support. 
Derives child context from simulated context, there is no need to preserve & rotate back parent handle, focusses on multi tcb info fields validation.